// @aws-sdk/util-create-request/dist-cjs/foo.fixture.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.httpRequest = exports.operationCommand = exports.fooClient = void 0;
const middleware_stack_1 = require("@aws-sdk/middleware-stack");
const protocol_http_1 = require("@aws-sdk/protocol-http");
const output = { Data: "data", $metadata: {} };
const input = { String: "input" };
exports.fooClient = {
    config: {},
    middlewareStack: (0, middleware_stack_1.constructStack)(),
    send: (command) => command.resolveMiddleware(exports.fooClient.middlewareStack, exports.fooClient.config, undefined)({ input }),
    destroy: () => { },
};
exports.operationCommand = {
    middlewareStack: (0, middleware_stack_1.constructStack)(),
    input: {},
    resolveMiddleware: (stack) => {
        const concatStack = stack.concat(exports.operationCommand.middlewareStack);
        return concatStack.resolve(() => Promise.resolve({ output, response: {} }), {});
    },
};
exports.httpRequest = new protocol_http_1.HttpRequest({
    protocol: "https:",
    path: "/foo",
    hostname: "foo-service.us-east-1.amazonaws.com",
    headers: {},
    method: "GET",
    body: "",
});
